controller/controller.go

```diff
@@ -121,7 +121,7 @@ func (fw *logEcapsulator) Write(p []byte) (n int, err error) {
 //go:embed static
 var static embed.FS
 
-func NewWithDependencies(listener, graphqlListener net.Listener, gqlToken string, recorder metrics.MetricsRecorder, backend state.State) (*Controller, error) {
+func NewWithDependencies(ctx *cli.Context, listener, graphqlListener net.Listener, recorder metrics.MetricsRecorder, backend state.State) (*Controller, error) {
```

Comment:
Nit: the purpose of seperating New from NewWithDependencies was to put the CLI parsing in a single place, so that tests wouldn't worry about assembling a CLI. Technically, NewWithDepedencies could be a private function and the tests could skip the _test underscore in the package name